### PR TITLE
Update NACHO modules from nf core

### PR DESCRIPTION
**Pull request by alanmmobbs93**
Closes [Nanostring #107 ](https://github.com/nf-core/nanostring/issues/107)
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/nanostring/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/nanostring _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

## Description
Modules moved from local to nf-core:
 - NACHO_NORMALIZE
 - NACHO_QC
 
Channels modified to contain meta field
Removed duplicated entries from nextflow.config

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -5,6 +5,7 @@
     Default config options for all compute environments
 ----------------------------------------------------------------------------------------
 */
+nextflow.enable.moduleBinaries = true
```
Comment (alanmmobbs93) [inline nextflow.config:8]: duplicated

File: subworkflows/local/compute_gene_scores_heatmap/main.nf [hunk 2]
```diff
@@ -9,12 +9,12 @@ include { COMPUTE_GENE_SCORES     } from '../../../modules/local/compute_gene_sc
 include { CREATE_GENE_HEATMAP     } from '../../../modules/local/create_gene_heatmap'
 
 workflow COMPUTE_GENE_SCORES_HEATMAP {
+
     take:
     normalized_counts          // channel: [ meta, normalized_counts.tsv ]
     annotated_endo_data        // channel: [ meta, annotated_endo_data.tsv ]
     ch_gene_score_yaml         // channel: [ file(gene_score_yaml.yaml) ]
     ch_heatmap_genes_to_filter // channel: [ file(heatmap_genes_to_filter.yaml) ]
-    skip_heatmap               // boolean
```
Comment (nschcolnicov) [inline subworkflows/local/compute_gene_scores_heatmap/main.nf:17]: Please revert this change, the subworkflow shouldn't get params being passed directly into it, that is why we are passing the params.skip_heatmap

File: subworkflows/local/compute_gene_scores_heatmap/main.nf [hunk 3]
```diff
@@ -28,19 +28,19 @@ workflow COMPUTE_GENE_SCORES_HEATMAP {
         ch_gene_score_yaml
     )
     ch_versions      = ch_versions.mix(COMPUTE_GENE_SCORES.out.versions)
-    ch_multiqc_files = ch_multiqc_files.mix(COMPUTE_GENE_SCORES.out.scores_for_mqc.map{ meta, file-> file }.collect())
+    ch_multiqc_files = ch_multiqc_files.mix(COMPUTE_GENE_SCORES.out.scores_for_mqc.map{ meta, file -> file }.collect())
 
     //
     // MODULE: Compute gene-count heatmap for MultiQC report based on annotated (ENDO) counts
     //
-    if(!skip_heatmap){
```
Comment (nschcolnicov) [inline subworkflows/local/compute_gene_scores_heatmap/main.nf:36]: Also revert here

File: subworkflows/local/quality_control/main.nf [hunk 4]
```diff
@@ -2,17 +2,17 @@
...
+    counts      // channel: [ meta, [rcc files] ]
+    samplesheet // channel: [ meta, path/to/samplesheet.csv ]
 
     main:
     NACHO_QC ( counts, samplesheet )
 
     emit:
-    nacho_qc_multiqc_metrics = NACHO_QC.out.nacho_qc_multiqc_metrics // channel: [ .png and .txt mqc files ]
-    versions                 = NACHO_QC.out.versions                 // channel: [ versions.yml ]
+    nacho_qc_multiqc_metrics = NACHO_QC.out.nacho_qc_png.map{it[1]}.mix(NACHO_QC.out.nacho_qc_txt.map{it[1]}) // channel: [ .png and .txt mqc files ]
```
Comment (nschcolnicov) [inline subworkflows/local/quality_control/main.nf:16]: Better to use join here instead of mix, this way all only the outputs that have the same meta will be merged together. This doesn't impact the pipeline now because we don't allow to run more than one sample at a time, and all of the files get mixed together.

File: workflows/nanostring.nf [hunk 5]
```diff
@@ -83,31 +87,26 @@ workflow NANOSTRING {
...
+    ch_versions         = ch_versions.mix(NORMALIZE.out.versions)
+    ch_normalized       = NORMALIZE.out.normalized_counts
+    ch_normalized_wo_hk = NORMALIZE.out.normalized_counts_wo_HK
 
     //
     // MODULE: Annotate normalized counts with metadata from the samplesheet
     //
     CREATE_ANNOTATED_TABLES (
-        NORMALIZE.out.normalized_counts.mix(NORMALIZE.out.normalized_counts_wo_HK).toSortedList().flatten(),
+        ch_normalized.mix(ch_normalized_wo_hk).toSortedList{ a, b -> a[1].getName() <=> b[1].getName() }.flatMap(), // Order tuples (based on file name) to stabilize tests and reproducibility
```
Comment (nschcolnicov) [inline workflows/nanostring.nf:98]: Same here, I would prefer we use join, even though it doesn't affect the current state of the pipeline, it may cause issues in the future.

**Review by nschcolnicov — APPROVED** [review]
